inventory-service/src/main/java/com/spaghettininjas/yaposs/service/InventoriesService.java

```diff
@@ -0,0 +1,57 @@
+package com.spaghettininjas.yaposs.service;
+
+import com.spaghettininjas.yaposs.repository.InventoryRepository;
+import com.spaghettininjas.yaposs.repository.ProductRepository;
+import com.spaghettininjas.yaposs.repository.entity.Inventory;
+import com.spaghettininjas.yaposs.repository.entity.InventoryDTO;
+import com.spaghettininjas.yaposs.repository.entity.Product;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+
+@Service
+public class InventoriesService {
+    public final InventoryRepository inventoryRepository;
+    public final ProductRepository productRepository;
+
+    public InventoriesService(InventoryRepository inventoryRepository, ProductRepository productRepository) {
+        this.inventoryRepository = inventoryRepository;
+        this.productRepository = productRepository;
+    }
+
+    public Optional<Inventory> findById(Long id){
+        return inventoryRepository.findById(id);
+    }
+
+    public void deleteById(Long id){
+        inventoryRepository.deleteById(id);
+    }
+
+    public Inventory save(InventoryDTO inventory) {
+        Long productId = inventory.getProductId();
+
+
+
+        Inventory newInventory = new Inventory();
+        newInventory.setStockQuantity(inventory.getStockQuantity());
+
+        Product product = productRepository.findById(productId)
+                .orElseThrow(() -> new RuntimeException("Product not found"));
```

Comment:
I think, this will probably return `500` on a bad request, it would be better to with `4XX` status code. `400 Bad Request` seems like the best fit.

Also some tests cases to try out error path would be nice.